test/unit/v1beta1/suggestion/test_optuna_service.py

```diff
@@ -184,280 +185,99 @@ def test_get_suggestion(self, algorithm_name, algorithm_settings):
-                algorithm_name="cmaes",
-                algorithm_settings=[
-                    api_pb2.AlgorithmSetting(
-                        name="restart_strategy",
-                        value="None",
-                    ),
-                    api_pb2.AlgorithmSetting(
-                        name="sigma",
-                        value="-10",
-                    ),
-                ],
-            )
-        )
-        _, _, code, _ = utils.call_validate(self.test_server, experiment_spec)
-        assert code == grpc.StatusCode.INVALID_ARGUMENT
-
-        # [CMAES] Invalid random_state
-        experiment_spec = api_pb2.ExperimentSpec(
-            algorithm=api_pb2.AlgorithmSpec(
-                algorithm_name="cmaes",
-                algorithm_settings=[
-                    api_pb2.AlgorithmSetting(
-                        name="sigma",
-                        value="0.2",
-                    ),
-                    api_pb2.AlgorithmSetting(
-                        name="random_state",
-                        value="-20",
-                    ),
-                ],
-            )
-        )
-        _, _, code, _ = utils.call_validate(self.test_server, experiment_spec)
-        assert code == grpc.StatusCode.INVALID_ARGUMENT
-
-        # [CMAES] Invalid number of parameters
-        experiment_spec = api_pb2.ExperimentSpec(
-            algorithm=api_pb2.AlgorithmSpec(
-                algorithm_name="cmaes",
-                algorithm_settings=[
-                    api_pb2.AlgorithmSetting(
-                        name="sigma",
-                        value="0.2",
-                    ),
-                ],
-            )
-        )
-        _, _, code, _ = utils.call_validate(self.test_server, experiment_spec)
-        assert code == grpc.StatusCode.INVALID_ARGUMENT
-
-        # [RANDOM] Valid Case
-        experiment_spec = api_pb2.ExperimentSpec(
-            algorithm=api_pb2.AlgorithmSpec(
-                algorithm_name="random",
-                algorithm_settings=[
-                    api_pb2.AlgorithmSetting(
-                        name="random_state",
-                        value="10",
-                    ),
-                ],
-            )
-        )
-        _, _, code, _ = utils.call_validate(self.test_server, experiment_spec)
-        assert code == grpc.StatusCode.OK
-
-        # [RANDOM] Invalid parameter name
-        experiment_spec = api_pb2.ExperimentSpec(
-            algorithm=api_pb2.AlgorithmSpec(
-                algorithm_name="random",
-                algorithm_settings=[
-                    api_pb2.AlgorithmSetting(
-                        name="invalid",
-                        value="invalid",
-                    ),
-                ],
-            )
-        )
-        _, _, code, _ = utils.call_validate(self.test_server, experiment_spec)
-        assert code == grpc.StatusCode.INVALID_ARGUMENT
-
-        # [RANDOM] Invalid random_state
+    @pytest.mark.parametrize(
```

Comment:
IMO, we would be good to do refactoring unit tests using `@pytest.mark.parametrize` in other suggestion services, same as this.

We can follow up with other PRs.

Reply:
I like this feature of PyTest, thanks!